### PR TITLE
Srtp kdf optest

### PR DESCRIPTION
**Pull request by kaleb-himes**
# Description

Exposing a service API for the new submission effort (module v6.0.0) makes these checks now being exposed passable while running in FIPS mode.

# Testing

Using the operational test app

# Checklist

 - [X] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by kaleb-himes** [comment]
NOTE: Do not merge this PR before merging https://github.com/wolfSSL/wolfssl/pull/7462 (not sure what is taking so long on the other one?) This PR contains some of the same commits as work is progressing faster than merges are happening.

**Comment by kaleb-himes** [comment]
rebased on master.

**Comment by kaleb-himes** [comment]
retest this please

File: wolfcrypt/src/pwdbased.c [hunk 1]
```diff
@@ -209,6 +213,24 @@ int wc_PBKDF2_ex(byte* output, const byte* passwd, int pLen, const byte* salt,
...
+        return BAD_LENGTH_E;
+#endif
+
+#if FIPS_VERSION3_GE(6,0,0) && defined(DEBUG_WOLFSSL)
+    /* SP800-132 section 5.2 recommends an iteration count of 1000 but this is
+     * not strictly enforceable and is listed in Appendix B Table 1 as a
+     * non-testable requirement. wolfCrypt will log it when appropriate but
+     * take no action */
+    if (iterations < 1000) {
+        WOLFSSL_MSG("WARNING: Iteration < 1,000, see SP800-132 section 5.2");
```
Comment (dgarske) [inline wolfcrypt/src/pwdbased.c:231]: Is this "warning" sufficient? Won't be noticed unless DEBUG_WOLFSSL is set. Should it return a new error code?

File: wolfssl/wolfcrypt/hmac.h [hunk 2]
```diff
@@ -43,6 +43,10 @@
     WOLFSSL_LOCAL int wolfCrypt_FIPS_HMAC_sanity(void);
 #endif
 
+#if FIPS_VERSION3_GE(6,0,0)
+    #define FIPS_ALLOW_SHORT 1
```
Comment (dgarske) [inline wolfssl/wolfcrypt/hmac.h:47]: Should this new FIPS macro include HMAC in the name? Would be nice if `wc_HmacSetKey_ex` was unsigned and actually a flag / bit mask. Currently its 0 or non-zero only.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/pwdbased.c (hunk 1 above)
Comment (kaleb-himes) [reply wolfcrypt/src/pwdbased.c:231]: Yes this is sufficient per the comment this is not enforceable.

File: wolfssl/wolfcrypt/hmac.h (hunk 2 above)
Comment (kaleb-himes) [reply wolfssl/wolfcrypt/hmac.h:47]: By design.

**Review by dgarske — APPROVED** [review]

**Pull request merged**